profile/src/wallet_kit_common/error/common_error.rs

```diff
@@ -0,0 +1,292 @@
+use crate::prelude::*;
+
+use thiserror::Error as ThisError;
+
+pub type Result<T, E = CommonError> = std::result::Result<T, E>;
+
+#[repr(u32)]
+#[derive(Clone, Debug, ThisError, PartialEq, uniffi::Error)]
+#[uniffi(flat_error)]
+pub enum CommonError {
```

Comment:
This error enum is by no means final, it is very very WIP... we would need to regroup and catagorize all errors. And probably have a plan for error handling in wallets first....